src/agent/multitype_queue/src/sqlitestorage.cpp

```diff
@@ -105,71 +114,90 @@ int SQLiteStorage::Store(const nlohmann::json& message, const std::string& table
 }
 
 // TODO: we shouldn't use rowid outside the table itself
-nlohmann::json SQLiteStorage::Retrieve(int id, const std::string& tableName, const std::string& moduleName)
+nlohmann::json SQLiteStorage::Retrieve(int id,
+                                       const std::string& tableName,
+                                       const std::string& moduleName,
+                                       [[maybe_unused]] const std::string& moduleType)
 {
-
     std::string selectQuery;
     if (moduleName.empty())
     {
-        constexpr std::string_view SELECT_QUERY {"SELECT module, message FROM {} WHERE rowid = ?;"};
+        constexpr std::string_view SELECT_QUERY {"SELECT module_name, module_type, metadata, message FROM {} WHERE "
+                                                 "rowid = ?;"};
         selectQuery = fmt::format(SELECT_QUERY, tableName);
     }
     else
     {
         constexpr std::string_view SELECT_QUERY {
-            "SELECT module, message FROM {} WHERE module LIKE \"{}\" AND rowid = ?;"};
+            "SELECT module_name, module_type, metadata, message FROM {} WHERE module_name LIKE \"{}\" AND rowid = ?;"};
```

Comment:
Don't you need to filter also with the `moduleType`?

Reply:
In this first stage, because of the way events are being sent, it is not necessary, but it should be done in the future.